### PR TITLE
Fixes for building on latest NDK

### PR DESCRIPTION
**Pull request by guusw**
I encountered some issues when building for android:
- On newer versions of clang `-Werror=declaration-after-statement` triggers even when compiling against C99 or later
- Editable/Selection are not imported inside SDLActivity and causes the file to not compile

Alternatively the files not following the `declaration-after-statement` rule could be fixed but I don't have time to do so, so this just fixed the android build

### CODE REVIEW

**Comment by slouken** [comment]
I fixed the missing imports. I'll update the NDK and look at fixing the source triggering the declaration after statement warning. Thanks!

**Comment by mingmoe** [comment]
There are some fixing patch.
https://github.com/libsdl-org/SDL/issues/5950

File: android-project/app/src/main/java/org/libsdl/app/SDLActivity.java [hunk 1]
```diff
@@ -53,6 +53,8 @@
 import android.widget.RelativeLayout;
 import android.widget.TextView;
 import android.widget.Toast;
+import android.text.Editable;
+import android.text.Selection;
```
Comment (madebr) [inline android-project/app/src/main/java/org/libsdl/app/SDLActivity.java:57]: Can you add these in alphabetic order?
They should be added before and after `import android.text.InputType;`.

**Comment by madebr** [comment]
While applying some fixes to CMake and trying out #5929, I accidentally pushed a commit to main bumping the android ndk to r25.
It causes CI to fail.
What should I do? Revert 6fa7d62408024e8de64547e33bbc7fbf25ad59ec or wait for fixing it?

**Comment by slouken** [comment]
Please revert both of those changes. I don't want to bump the SDK or platform version at this point.

**Comment by madebr** [comment]
Okay, reverted in https://github.com/libsdl-org/SDL/commit/2fdedd17be03678c89e65bc2bd0672b48f38c5e9

**Comment by slouken** [comment]
I think both of these issues are taken care of now. @guusw, can you confirm?

**Comment by guusw** [comment]
Will check

**Comment by guusw** [comment]
Java files compile and can build with ndk-25, thanks. Closing this as fixed

**Pull request closed without merging**